### PR TITLE
Update details in the book

### PR DESCRIPTION
**Pull request by veebs**

### CODE REVIEW

File: src/architecture.xml [hunk 1]
```diff
@@ -78,8 +82,60 @@
     </para>
     <para>
       For more information, please refer to the
-      <link xlink:href="&API;buffer/package-summary.html#package_description"><literal>org.jboss.netty.buffer</literal> package description</link>.
+      <link xlink:href="&API;buffer/package-summary.html#package_description">org.jboss.netty.buffer</link>
```
Comment (trustin) [inline src/architecture.xml]: Any reason why the `<literal>` tag has been removed?

File: src/architecture.xml [hunk 2]
```diff
@@ -78,8 +82,60 @@
     </para>
     <para>
       For more information, please refer to the
-      <link xlink:href="&API;buffer/package-summary.html#package_description"><literal>org.jboss.netty.buffer</literal> package description</link>.
+      <link xlink:href="&API;buffer/package-summary.html#package_description">org.jboss.netty.buffer</link> 
+      package description.
     </para>
+    <section>
+      <title>Zero Copy</title>
```
Comment (trustin) [inline src/architecture.xml]: Due to the way Java manages direct buffer, it's practically not easy to use direct buffers directly in most case, and thus it's not easy for a user to achieve zero-copy (impossible for most cases).  Therefore, I'd drop this section until we make some progress on this topic.

File: src/architecture.xml [hunk 3]
```diff
@@ -216,7 +272,7 @@
     </para>
 
     <section>
-      <title>Codec framework</title>
+      <title>Codec Framework</title>
```
Comment (trustin) [inline src/architecture.xml]: I prefer not to capitalize non-first words.  We can always capitalized them using CSS.

File: src/faq.xml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+-->
+<!DOCTYPE preface SYSTEM "../lib/schema-5.1b4/dtd/docbook.dtd" [
+<!ENTITY % Entities SYSTEM "dtds/entities.dtd">
+%Entities;
+]>
+<preface xmlns="http://docbook.org/ns/docbook"
+         xml:id="faq">
+  <title>Frequently Asked Questions</title>
```
Comment (trustin) [inline src/faq.xml:23]: Wouldn't it be better just add a link to [our StackSverflow.com page](http://stackoverflow.com/questions/tagged/netty?sort=faq)?  If there's a good reason, I'm fine with keeping this section while adding the link to the StackOverflow.com FAQ page at the top of the section.

File: src/architecture.xml (hunk 1 above)
Comment (veebs) [inline src/architecture.xml]: Not sure why it was removed.  Maybe auto-format.  I'll put it back.

File: src/architecture.xml (hunk 2 above)
Comment (veebs) [inline src/architecture.xml]: OK.

File: src/architecture.xml (hunk 3 above)
Comment (veebs) [inline src/architecture.xml]: OK

File: src/faq.xml (hunk 4 above)
Comment (veebs) [inline src/faq.xml:23]: I though I would add this section as a "best off" and concise summary of our StackOverflow.

I'll add the link at the top of the page.

**Comment by trustin** [comment]
Awesome.  Thanks a lot!  Did you deploy the new version to netty-website?

**Comment by veebs** [comment]
No - I just committed the updated changes.

I can help deploy if you are busy.  I do need some pointers/instructions
because I've not deployed on github before.

Regards
Vibul

On 28 March 2012 16:46, Trustin Lee <
reply@reply.github.com

**Comment by trustin** [comment]
It's as simple as the following:

```
$ cd netty-book
$ ./build.sh
```

Then you will find the generated documents in the target directory.  You can copy then to netty-website and deploy as usual, by executing _bin/deploy.sh

However, after merging this pull request, I see some warnings and errors in the build output, such as namespace warnings and missing image files.  We need to fix them before deploying anything.

**Comment by veebs** [comment]
No problems.

I'll look at fixing the warnings tomorrow and then deploying.

V

On 28 March 2012 17:06, Trustin Lee <
reply@reply.github.com

**Comment by veebs** [comment]
Hi Trustin,

I have it a shot and spent the last hour looking at the problem.

Unfortunately, I'm no closer.  Any hints on how to fix these issues?

Regards
Vibul

**Comment by trustin** [comment]
First thing to do is to add the missing image files to the repository. :-)

**Comment by veebs** [comment]
Sorry about that. Forgot to add them to git.  That's why I did not see the
errors when I ran build locally.

The images have been merged now.

Vibul.

On 29 March 2012 22:05, Trustin Lee <
reply@reply.github.com

**Pull request merged**